packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -427,6 +427,21 @@ export class DataStores implements IDisposable {
             throw responseToException(create404Response(request), request);
         }
 
+        if (context.tombstoned) {
+            // The requested data store is tombstoned by gc. Throw a 404 response exception.
+            const request = { url: id };
+            const urlNoQuery = request.url.split("?")[0];
+            const errWithStack = generateErrorWithStack();
+
+            const response: IResponse = {
+                mimeType: "text/plain",
+                status: 404,
+                value: `Datastore tombstoned: ${urlNoQuery}`,
```

Comment:
I think even after we implement full sweep, we may refer against the list of Deleted DataStores to give a distinct error here (v. above where it's just not found with no further context).  So I would want this message to not mention Tombstone, but say "Datastore removed by GC" or something like that.

@agarwal-navin this is what we were just talking about

Reply:
Ok, I can update the message